### PR TITLE
Test travis optim

### PR DESCRIPTION
**Pull request by jamesbhobbs**
Cuts build time

Makes no change to actual tests run, just how they are configured to run in parallel...

Changes:
1. Remove all caching (I tested at the beginning and test of the commits in this branch and reran tests at least twice, consistently without caching was faster on even non-first builds).
2. Move install script to be inline of the first test build stage (it was used in three before, now two of them no longer exist (removed in point 6 of this list).
3. Move the code-climate test coverage reporter to be inline so we get reports earlier (slightly out of scope). NB that coverage is reported on this branch is the only test this needs.
4. Move `xvfb-run npm run test:e2e:ci` and `xvfb-run npm run test:accessibility:ci` to prod build stage (the second stage called test which has an `npm install --production` rather than `npm ci`).
5. Move `xvfb-run npm run test:e2e:storybook:ci` to the first build stage (as it can never run on the production build).
6. Removed 2 build stages which are now unused (this means 2 fewer VMs are used - out of what was 4 in total)
7. Update in package.json the scripts to install bbc-a11y and cypress only after the production build successfully runs with the `npm install --production`.

Test time is from 6m5s-6m45s in real time with a total (parallel) runtime of 11-12m. So a saving in real time of 0 when all 4 builds start at the same time (which is rare), and a saving of the full wait time for the VMs to start each stage plus the saving of 2 VMs so it occupies half as many VM queue positions (therefore the queue of builds is halved) - meaning there is a real time saving in almost all cases and in the worst case there is no change. And the total runtime saving is approx 10 minutes.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,29 +1,23 @@
...
-       - xvfb-run npm run test:accessibility:ci
-      after_script: ./cc-test-reporter after-build -t lcov --debug --exit-code $TRAVIS_TEST_RESULT
-    - stage: test
-      script: xvfb-run npm run test:e2e:ci
-    - stage: test
-      script: xvfb-run npm run test:e2e:storybook:ci
+       - ./cc-test-reporter after-build -t lcov --debug --exit-code $TRAVIS_TEST_RESULT
+       - xvfb-run npm run test:e2e:storybook:ci
     - stage: test
       install: npm install --production
```
Comment (jtart) [inline .travis.yml]: We could also use `npm ci --only=prod` here.

**Review by BogdanDogaru — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -25,10 +25,10 @@
     "storybook": "start-storybook -p 9001 -c .storybook",
     "test": "npm run test:lint && npm run test:unit && npm run dataValidate",
     "test:accessibility": "npm run killApp && npm run build && run-p --race start bbcA11y",
-    "test:accessibility:ci": "npm run build && run-p --race start bbcA11y",
+    "test:accessibility:ci": "npm install bbc-a11y && run-p --race start bbcA11y",
```
Comment (sareh) [inline package.json]: The scripts `npm run test:accessibility:ci` and `npm run test:e2e:ci` won't work locally like this, and their only use is on Travis, so I suggest replacing these with a single script that runs this:

`"npm run build && npm install cypress bbc-a11y && run-p --race start cypress && run-p --race start bbcA11y",`

Then in the Travis config, there will just be a single script run for the production test.

**Review by jtart — CHANGES_REQUESTED** [review]
My change might only happen is we make a similar change in the jenkins build.

File: package.json (hunk 2 above)
Comment (jamesbhobbs) [reply package.json]: This is fine BUT the npm run build should be distinct as it's actually testing that the production build can build it isn't really a part of the remainder of the tests, though it is a prerequisite.

**Review by sareh — DISMISSED** [review]
👍 Looks good.

File: .travis.yml [hunk 3]
```diff
@@ -1,29 +1,22 @@
...
+        - ./cc-test-reporter after-build -t lcov --debug --exit-code $TRAVIS_TEST_RESULT
+        - xvfb-run npm run test:e2e:storybook:ci
     - stage: test
-      script: xvfb-run npm run test:e2e:ci
-    - stage: test
-      script: xvfb-run npm run test:e2e:storybook:ci
-    - stage: test
-      install: npm install --production
-      script: npm run build
+      install: npm ci only=production
```
Comment (sareh) [inline .travis.yml]: The syntax is `npm ci --only=production` https://docs.npmjs.com/cli/install

File: package.json [hunk 4]
```diff
@@ -25,10 +25,9 @@
     "storybook": "start-storybook -p 9001 -c .storybook",
     "test": "npm run test:lint && npm run test:unit && npm run dataValidate",
     "test:accessibility": "npm run killApp && npm run build && run-p --race start bbcA11y",
-    "test:accessibility:ci": "npm run build && run-p --race start bbcA11y",
     "test:ampValidate": "npm run killApp && npm run build && run-p --race start amp:validate",
+    "test:ci": "npm install cypress bbc-a11y && run-p --race start cypress && run-p --race start bbcA11y",
```
Comment (sareh) [inline package.json]: This should be updated to `npx install cypress bbc-a11y`, so the installation doesn't update the package.json.

File: .travis.yml (hunk 3 above)
Comment (jamesbhobbs) [reply .travis.yml]: Done

File: package.json (hunk 4 above)
Comment (sareh) [reply package.json]: Sorry, I was mistaken in that above comment.
`npm install --no-save cypress bbc-a11y` should be sufficient.

**Pull request merged**